### PR TITLE
Default basis_dir

### PR DESCRIPTION
**Pull request by MikhailRyazanov**
Implementation of a reasonable default directory (instead of `'.'`) for storing basis sets (see #325)...

Seems to work and pass tests at least in Linux and Windows, hope to see the same in macOS after @danhickstein enables it (#323).

(This PR can be merged before or after #326, but after merging one, the other needs to be adjusted accordingly.)

### CODE REVIEW

**Comment by DanHickstein** [comment]
This seems like the right move. Great work Mikhail!  I'll try to get MacOS testing working and review these changes next week.

File: README.rst [hunk 1]
```diff
@@ -71,6 +71,17 @@ Or, if you wish to edit the PyAbel source code without re-installing each time :
...
     python setup.py develop
 
+Before uninstalling
+~~~~~~~~~~~~~~~~~~~
+
+Some transform methods can save generated basis sets to disk. If you want to uninstall PyAbel completely, these files need to be removed as well. To do so, please *first* run the following script::
+
+    import abel
+    import shutil
+    shutil.rmtree(abel.transform.get_basis_dir())
```
Comment (stggh) [inline README.rst:81]: neat!

**Review by stggh — APPROVED** [review]
A lot of work for a small directory change. Good stuff. 

Works well under Linux (Ubuntu 21.04), and Mac OSX 11.5.2.

**Review by DanHickstein — APPROVED** [review]
Awesome work! This is a long-needed feature for PyAbel. This has the potential to massively speed up the workflow of certain users. The only issue I see here is if we have outdated basis sets floating that cause compatibility issues when we update the software. But, we had that problem anyways, so there's no reason not to get this implemented now. I'd say go ahead and merge whenever you like!

**Pull request merged**